app/javascript/packs/webauthn-setup.js

```diff
@@ -0,0 +1,57 @@
+function webauthn() {
+  const arrayBufferToBase64 = function(buffer) {
+    let binary = '';
+    const bytes = new Uint8Array(buffer);
+    const len = bytes.byteLength;
+    for (let i = 0; i < len; i += 1) {
+      binary += String.fromCharCode(bytes[i]);
+    }
+    return window.btoa(binary);
+  };
+  const longToByteArray = function(long) {
+    const byteArray = new Uint8Array(8);
+    for (let index = 0; index < byteArray.length; index += 1) {
+      const byte = long & 0xff;  // eslint-disable-line no-bitwise
+      byteArray[index] = byte;
+      long = (long - byte) / 256;
+    }
+    return byteArray;
+  };
+  const userId = document.getElementById('user_id').value;
+  const userEmail = document.getElementById('user_email').value;
+  const challengeBytes = new Uint8Array(JSON.parse(document.getElementById('user_challenge').value));
+  const createOptions = {
+    publicKey: {
+      challenge: challengeBytes,
+      rp: { name: window.location.hostname },
+      user: {
+        id: longToByteArray(userId),
+        name: userEmail,
+        displayName: userEmail,
+      },
+      pubKeyCredParams: [
+        {
+          type: 'public-key',
+          alg: -7,
+        },
+        {
+          type: 'public-key',
+          alg: -257,
```

Comment:
This is a temporary assignment that expires on 19 Apr 2019 (https://www.iana.org/assignments/cose/cose.xhtml#algorithms referenced from the WebAuthn recommendation)